client/me/memberships/subscription.jsx

```diff
@@ -33,42 +33,67 @@ function Subscription( { translate, subscription, moment, stoppingStatus } ) {
 	useEffect( () => {
 		if ( stoppingStatus === 'fail' ) {
 			// run is-error notice to contact support
-			dispatch(
-				errorNotice(
-					translate(
-						'There was a problem while stopping your subscription, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
-						{
-							components: {
-								a: <a href={ CALYPSO_CONTACT } />,
-								strong: <strong />,
-							},
-						}
-					)
-				)
-			);
+			subscription && ! subscription.renew_interval
+				? dispatch(
+						errorNotice(
+							translate(
+								'There was a problem while removing your product, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
+								{
+									components: {
+										a: <a href={ CALYPSO_CONTACT } />,
+										strong: <strong />,
+									},
+								}
+							)
+						)
+				  )
+				: dispatch(
+						errorNotice(
+							translate(
+								'There was a problem while stopping your subscription, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
+								{
+									components: {
+										a: <a href={ CALYPSO_CONTACT } />,
+										strong: <strong />,
+									},
+								}
+							)
+						)
+				  );
 		} else if ( stoppingStatus === 'success' ) {
 			// redirect back to Purchases list
 			dispatch(
-				successNotice(
-					translate( 'This subscription has been canceled. You will no longer be charged.' ),
-					{ displayOnNextPage: true }
-				)
+				successNotice( translate( 'This item has been removed.' ), { displayOnNextPage: true } )
 			);
 			page( purchasesRoot );
 		}
 	}, [ stoppingStatus, dispatch, translate ] );
```

Comment:
Let's add `subscription` (or `isProduct` if we go with Miguel's [suggestion](https://github.com/Automattic/wp-calypso/pull/62604/files#r845435267)) to the `useEffect`'s dependencies array.